Handbook: New beta and roadmap guidelines (#12295)
PostHog/posthog.com: contents/handbook/product/releasing-as-beta.md

---
title: Releasing early access and coming soon features
sidebar: Handbook
showTitle: true
hideAnchor: true
---

> PostHog includes a variety of early access features [the feature previews section of a users' settings page](https://app.posthog.com/settings/user-feature-previews), as well as a roadmap of feature previews which are coming soon. 

Reviewer: ```suggestion
PostHog includes a variety of early access features in [the feature previews section of a users' settings page](https://app.posthog.com/settings/user-feature-previews), as well as a roadmap of feature previews which are coming soon.
```

Items in the feature previews section can be toggled on or off if users want to try a feature out. Items in the coming soon section enable users to register their interest so that we can contact them with updates. Both sections work only at the user-level and not at the org or project level. 

## Releasing as coming soon

> If your team is considering building a feature in the future, you should consider adding it to the coming soon menu. This responsibility usually sits with the Team Lead, unless the project is an individual one. 

Reviewer: ```suggestion
If your team is considering building a feature in the future, you should add it to the coming soon menu. This responsibility usually sits with the Team Lead, unless the project is an individual one. 
```

> To add an item to the coming soon, simply create a feature flag with the `concept` state in [PostHog's early access features tool](/docs/feature-flags/early-access-feature-management).

Reviewer: ```suggestion
To add an item to the coming soon, simply create a feature flag with the `concept` state in PostHog's <PrivateLink url="https://us.posthog.com/project/2/early_access_features">early access features tool</PrivateLink>.
```

Adding items to the coming soon menu early offers several advantages. It enables us to gauge interest in a new feature via sign-ups, equips our marketing teams with news they can promote to users, and ensures that betas can have sample users ready from the moment they launch. 

Coming soon features can either be large or small, so use your judgement about what is of interest to users, but it should be something that you expect to work on in the next 3-6 months. 

> > **Moving from `Concept` to `Beta`**
> > Once you are ready to move an item from the coming soon roadmap to a beta which users can interact with, simply update the flag state from `concept` to `beta`. This will trigger an automatic notification to all subscribed users letting them know that they beta is available. All users who are subscribed to updates will be immediately opted in to the beta. 

Reviewer: ```suggestion
<CalloutBox icon="IconInfo" title="Moving from Concept to Beta" type="fyi">
  Once you are ready to move an item from the coming soon roadmap to a beta which users can interact with, simply update the flag state from `concept` to `beta`. This will trigger an automatic notification to all subscribed users letting them know that they beta is available. All users who are subscribed to updates will be immediately opted in to the beta. 
</CalloutBox>
```

/> 
<Caption>Betas should include a title, description, feedback button, and link to basic docs</Caption>

All betas should follow the best practices below in order to provide a minimum amount of information and usability for customers.

- Betas in the feature preview menu should include a title and short description 
- Betas in the feature preview menu should include a 'Give feedback' button
- Betas in the feature preview menu should have documentation (marked as beta) linked to them
- Betas should have a [feature owner](/handbook/engineering/feature-ownership)

Product teams are responsible for [writing documentation](/handbook/engineering/writing-docs), but the [Content team](/teams/content) can help, if needed. Titles, descriptions, and links can be added using [the early access menu](https://us.posthog.com/project/2/early_access_features).

> > **Launching a new beta?** It's helpful to let the Marketing teams know when new betas are added. They'll then add the beta to [the changelog](https://posthog.com/changelog/), organize any marketing announcements, plan [a full announcement](https://github.com/PostHog/meta/issues/new?template=launch-plan-.md) for full release, create an email onboarding flow to help you collect user feedback, and anything else you need. You can let them know via [the Marketing Slack channel](https://posthog.slack.com/archives/C08CG24E3SR).

Reviewer: ```suggestion
<CalloutBox icon="IconInfo" title="Launching a new beta?" type="fyi">
  It's helpful to let the Marketing teams know when new betas are added. They'll then add the beta to [the changelog](https://posthog.com/changelog/), organize any marketing announcements, plan [a full announcement](https://github.com/PostHog/meta/issues/new?template=launch-plan-.md) for full release, create an email onboarding flow to help you collect user feedback, and anything else you need. You can let them know via [the Marketing Slack channel](https://posthog.slack.com/archives/C08CG24E3SR).
</CalloutBox>
```
